src/validation.cpp

```diff
@@ -3642,34 +3642,54 @@ static bool CheckBlockHeader(const CBlockHeader& block, BlockValidationState& st
 static bool CheckMerkleRoot(const CBlock& block, BlockValidationState& state)
 {
     bool mutated;
-    uint256 hashMerkleRoot2 = BlockMerkleRoot(block, &mutated);
-    if (block.hashMerkleRoot != hashMerkleRoot2)
-        return state.Invalid(BlockValidationResult::BLOCK_MUTATED, "bad-txnmrklroot", "hashMerkleRoot mismatch");
+    uint256 merkle_root = BlockMerkleRoot(block, &mutated);
+    if (block.hashMerkleRoot != merkle_root) {
+        return state.Invalid(
+            /*result=*/BlockValidationResult::BLOCK_MUTATED,
+            /*reject_reason=*/"bad-txnmrklroot",
+            /*debug_message=*/"hashMerkleRoot mismatch");
+    }
 
     // Check for merkle tree malleability (CVE-2012-2459): repeating sequences
     // of transactions in a block without affecting the merkle root of a block,
     // while still invalidating it.
-    if (mutated)
-        return state.Invalid(BlockValidationResult::BLOCK_MUTATED, "bad-txns-duplicate", "duplicate transaction");
+    if (mutated) {
+        return state.Invalid(
+            /*result=*/BlockValidationResult::BLOCK_MUTATED,
+            /*reject_reason=*/"bad-txns-duplicate",
+            /*debug_message=*/"duplicate transaction");
+    }
 
     return true;
 }
 
 static std::optional<bool> CheckWitnessCommitment(const CBlock& block, BlockValidationState& state)
 {
     int commitpos = GetWitnessCommitmentIndex(block);
-    if (commitpos != NO_WITNESS_COMMITMENT) {
-        bool malleated = false;
-        uint256 hashWitness = BlockWitnessMerkleRoot(block, &malleated);
+
+    bool has_witness_commitment{commitpos != NO_WITNESS_COMMITMENT};
```

Comment:
nit c84db8d60e87eabc1e4f88ef05bf06fbf4db6f02

```suggestion
    const bool has_witness_commitment{commitpos != NO_WITNESS_COMMITMENT};
```